smdebug/core/utils.py

```diff
@@ -49,18 +49,21 @@ class FRAMEWORK(Enum):
 _smddp_tf_imported = None
 _smddp_pt_imported = None
 _is_using_smmodelparallel = None
+_smp_imported = None
 
-try:
-    import smdistributed.modelparallel.tensorflow as smp
 
-    _smp_imported = smp
-except (ImportError, ModuleNotFoundError):
+if check_smmodelparallel_training():
     try:
-        import smdistributed.modelparallel.torch as smp
+        import smdistributed.modelparallel.tensorflow as smp
 
         _smp_imported = smp
     except (ImportError, ModuleNotFoundError):
-        _smp_imported = None
+        try:
```

Comment:
Instead of chaining the two try-except block, can we decouple them?

Currently, if `smp.tensorflow` failed to import, the code tries to import `smp.torch`.
However, if `smp.torch` fails to import with an error other than `(ImportError, ModuleNotFoundError)`, the stack trace will be a little confusing, which happened in this incident of COE (`PermissionError`).

In particular, two stack traces showed up in this COE incident:
1. try import `smp.tf` failed
2. when trying to solve the above, PermissionError shows up
    1. no except block caught this type of error
    2. no pytorch keyword mentioned on the error itself.

You can see this led to wrong investigation in a PyTorch scenario.